server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -192,7 +192,7 @@ void recoverFromLocalShards(
                     // just trigger a merge to do housekeeping on the
                     // copied segments - we will also see them in stats etc.
                     indexShard.getEngine().forceMerge(false, -1, false, false, false, UUIDs.randomBase64UUID());
-                    if (indexShard.isRemoteTranslogEnabled() && indexShard.shardRouting.primary()) {
+                    if ((indexShard.isRemoteTranslogEnabled() || indexShard.isMigratingToRemote()) && indexShard.shardRouting.primary()) {
```

Comment:
We wouldn't start  `LOCAL_SHARDS` recovery for indices which are migrating . Hence this is not needed . Likewise all changes in this file.

Reply:
Addressed